### PR TITLE
connectionRequestTimeout for httpClient Fixes gh-799

### PR DESCRIPTION
**Pull request by galaxy-sea**
Fixes issue [Can't config connection request timeout for http client #799](https://github.com/spring-cloud/spring-cloud-openfeign/issues/799)

### CODE REVIEW

**Comment by szpak** [comment]
I had bumped into that problem recently and I've just wanted to report that future request :-).

It is especially useful, as "total" timeout to perform the request-response pair doesn't seem to be supported by HC5 - https://issues.apache.org/jira/browse/HTTPCLIENT-1074

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/FeignHttpClientProperties.java [hunk 1]
```diff
@@ -199,6 +200,16 @@ public static class Hc5Properties {
 		 */
 		public static final TimeUnit DEFAULT_SOCKET_TIMEOUT_UNIT = TimeUnit.SECONDS;
 
+		/**
+		 * Default value for connection request timeout.
+		 */
+		public static final int DEFAULT_CONNECTION_REQUEST_TIMEOUT = 5;
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/FeignHttpClientProperties.java:206]: In order to maintain compatibility, let's stick with the HC5 defualt connection request timeout (3 minutes).

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @galaxy-sea, thanks for submitting this PR. I've added a comment - please address it. Also, please submit these changes against `3.1.x` instead of `main`, so we'll get them included in the `2021.x` release train as well.

**Comment by galaxy-sea** [comment]
> Hello @galaxy-sea, thanks for submitting this PR. I've added a comment - please address it. Also, please submit these changes against `3.1.x` instead of `main`, so we'll get them included in the `2021.x` release train as well.

hello @OlgaMaciaszek, I resubmitted a PR for the 3.1. x branch . [pr#817](https://github.com/spring-cloud/spring-cloud-openfeign/pull/817)

**Comment by OlgaMaciaszek** [comment]
Thanks @galaxy-sea . Closing in favour of https://github.com/spring-cloud/spring-cloud-openfeign/pull/817.

**Pull request closed without merging**